### PR TITLE
Migration to hibernate 6

### PR DESCRIPTION
**Pull request by danielvion**
Still pretty much a big WIP

### CODE REVIEW

File: querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/AbstractHibernateQuery.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ private Query createQuery(@Nullable QueryModifiers modifiers, boolean forCount)
       query.setLockMode(entry.getKey().toString(), entry.getValue());
     }
     if (flushMode != null) {
-      query.setFlushMode(flushMode);
+      query.setFlushMode(flushMode.toJpaFlushMode());
```
Comment (mira-silhavy) [inline querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/AbstractHibernateQuery.java:122]: Hi, could you replace this with `setHibernateFlushMode` instead?

**Comment by mira-silhavy** [comment]
Hi, how is this looking? We're eager to test this. We're currently using this fork https://github.com/marok39/querydsl, it's also patched to support Hibernate 6. Are you aware of that fork?

**Comment by velo** [comment]
> Hi, how is this looking? We're eager to test this. We're currently using this fork https://github.com/marok39/querydsl, it's also patched to support Hibernate 6. Are you aware of that fork?

No, I didn't. Will try to join forces

**Comment by velo** [comment]
> Hi, how is this looking? We're eager to test this. We're currently using this fork https://github.com/marok39/querydsl, it's also patched to support Hibernate 6. Are you aware of that fork?

Seems both are extremely similar
https://github.com/OpenFeign/querydsl/pull/135/files#

I didn't spot any significant changes.  For sure will bring in the changes to have all javax.* ported to jakarta.*, but other than that seems to be the same code.

**Pull request merged**